frontend/catalyst/debug/printing.py

```diff
@@ -132,7 +133,7 @@ def func(x: float):
 
     Outside a :func:`qjit` compiled function the operation falls back to the Python print statement.
     """
-    if EvaluationContext.is_tracing():
+    if capture_enabled() or EvaluationContext.is_tracing():
```

Comment:
Isn't `capture_enabled` a global flag, whereas `is_tracing` is a contextual one?

Reply:
Interesting, don't you need an impl for the print primitive as well then? The `print_memref` is a runtime print that doesn't use callbacks, so has it's own `print_p`.

Reply:
Looks like there was actually no pytests for `print_memref` outside a `qjit` context... 🤦‍♀️ I'll figure out what the behavior should be and add a test.

Reply:
>Though does it even make sense to be called outside of tracing since x must be a tracer?

That's why the dispatch to python exists. We want program's written for catalyst to still work even you remove the qjit decorator, this provides a lot of convenience to users :)

Reply:
Looks like print_p has an impl now, closing.